### ISSUE TITLE
Offline mode

### ISSUE THREAD

**Issue by NULLx76** [RobbinBaauw/CSHub#131]
**Is your feature request related to a problem? Please describe.**
Semi related to #45 , as this feature request will likely incorporate that issue.

Also related to #97 , however that likely will never be fixed as it is incompatible with our current real time editing feature.

**Describe the solution you'd like**
It would be nice if CSHub would work offline, with of course limited functionality. Just viewing posts should be good enough, if we really want to we could make it so that things like favouriting would still work and were "queued up" for when a connection becomes available again.  

**Additional context**
The easiest way to achieve this is probably by using the service worker, Google has good documentation on how to achieve this, which we can make use of.

**Comment by RobbinBaauw** [comment]
Yeah the odd thing is, it should already work: we use the service working and a lot of posts are cached (though we might need to more aggresive caching)

**Comment by NULLx76** [comment]
In that case consider this a bug report

**Comment by RobbinBaauw** [comment]
We could do this by taking the cache if no response within an x amount of seconds

**Comment by RobbinBaauw** [comment]
We should actually use a cache first approach!
